packages/stream_chat/lib/src/client/client.dart

```diff
@@ -2174,6 +2234,103 @@ class ClientState {
     );
   }
 
+  void _listenLocationShared() {
```

Comment:
The client and channel files are very large. I wonder if we for example can put all location related code to separates files.

Reply:
Yes, we should move all the event handling logic outside of state into another file but let's do it in a separate ticket.